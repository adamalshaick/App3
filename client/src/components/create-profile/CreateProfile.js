import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

class CreateProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displaySocialInputs: false,
      handle: "",
      company: "",
      website: ""
    };
  }

  render() {
    return <div />;
  }
}

export default connect(null)(CreateProfile);
