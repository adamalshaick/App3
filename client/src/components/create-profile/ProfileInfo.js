import React from "react";
import TextFieldGroup from "../common/TextFieldGroup";
import TextAreaFieldGroup from "../common/TextAreaFieldGroup";
import UploadFileGroup from "../common/UploadFileGroup";

const ProfileInfo = ({
  errors,
  handle,
  location,
  bio,
  onChange,
  fileSelectedHandler
}) => {
  return (
    <>
      <div className="row">
        <div className="col-md-6 mb-2">
          <UploadFileGroup
            error={errors.file}
            icon="fas fa-file-upload fa-8x"
            type="file"
            name="file"
            onChange={fileSelectedHandler}
            info="Profile Image"
          />
        </div>
        <div className="col-md-6 mt-5">
          <TextFieldGroup
            id="handle"
            placeholder="* Username"
            name="handle"
            value={handle}
            onChange={onChange}
            error={errors.handle}
            info="Your username"
          />
          <TextFieldGroup
            id="location"
            placeholder="Location"
            name="location"
            value={location}
            onChange={onChange}
            error={errors.location}
            info="Your Location"
          />
        </div>
      </div>
      <TextAreaFieldGroup
        id="bio"
        placeholder="Short Bio"
        name="bio"
        value={bio}
        onChange={onChange}
        error={errors.bio}
        info="Your bio"
      />
    </>
  );
};

export default ProfileInfo;
