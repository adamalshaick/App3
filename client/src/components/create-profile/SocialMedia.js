import React from "react";
import InputGroup from "../common/InputGroup";

const SocialMedia = ({
  twitter,
  facebook,
  linkedin,
  youtube,
  instagram,
  onChange,
  errors
}) => {
  return (
    <section>
      <InputGroup
        id="twitter"
        placeholder="Twitter Profile URL"
        name="twitter"
        icon="fab fa-twitter"
        value={twitter}
        onChange={onChange}
        error={errors.twitter}
      />

      <InputGroup
        id="facebook"
        placeholder="Facebook Page URL"
        name="facebook"
        icon="fab fa-facebook"
        value={facebook}
        onChange={onChange}
        error={errors.facebook}
      />

      <InputGroup
        id="linkedin"
        placeholder="Linkedin Profile URL"
        name="linkedin"
        icon="fab fa-linkedin"
        value={linkedin}
        onChange={onChange}
        error={errors.linkedin}
      />

      <InputGroup
        id="youtube"
        placeholder="YouTube Channel URL"
        name="youtube"
        icon="fab fa-youtube"
        value={youtube}
        onChange={onChange}
        error={errors.youtube}
      />

      <InputGroup
        id="instagram"
        placeholder="Instagram Page URL"
        name="instagram"
        icon="fab fa-instagram"
        value={instagram}
        onChange={onChange}
        error={errors.instagram}
      />
    </section>
  );
};

export default SocialMedia;
