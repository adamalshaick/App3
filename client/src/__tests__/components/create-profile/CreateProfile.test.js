import React from "react";
import { CreateProfile } from "../../../components/create-profile/CreateProfile";
import { shallow } from "enzyme";
import ProfileInfo from "../../../components/create-profile/ProfileInfo";
import SocialMedia from "../../../components/create-profile/SocialMedia";
import toJson from "enzyme-to-json";

const mockCreateProfilefn = jest.fn();
const mockGetCurrentProfilefn = jest.fn();
const Props = {
  profile: {},
  errors: {}
};

const wrapper = shallow(
  <CreateProfile
    {...Props}
    getCurrentProfile={mockGetCurrentProfilefn}
    createProfile={mockCreateProfilefn}
  />
);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("component did mount", () => {
  it("calls get profile fn", () => {
    expect(mockGetCurrentProfilefn.mock.calls.length).toEqual(1);
  });
});

describe("create profile action without displaying social inputs", () => {
  const profileData = new FormData();
  profileData.append("handle", "handle value");
  profileData.append("location", "location value");
  profileData.append("bio", "bio value");
  profileData.append("twitter", "");
  profileData.append("facebook", "");
  profileData.append("linkedin", "");
  profileData.append("youtube", "");
  profileData.append("instagram", "");

  it("doesn't display social inputs", () => {
    expect(wrapper.find(SocialMedia).length).toEqual(0);
  });

  it("calls function with correct data", () => {
    wrapper
      .find(ProfileInfo)
      .dive()
      .find("#handle")
      .simulate("change", {
        target: { name: "handle", value: "handle value" }
      });
    wrapper
      .find(ProfileInfo)
      .dive()
      .find("#location")
      .simulate("change", {
        target: { name: "location", value: "location value" }
      });
    wrapper
      .find(ProfileInfo)
      .dive()
      .find("#bio")
      .simulate("change", { target: { name: "bio", value: "bio value" } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(mockCreateProfilefn.mock.calls[0][0]).toEqual(profileData);
  });
});

describe("create profile action with social inputs", () => {
  const profileSocialData = new FormData();
  profileSocialData.append("handle", "handle value");
  profileSocialData.append("location", "location value");
  profileSocialData.append("bio", "bio value");
  profileSocialData.append("twitter", "twitter");
  profileSocialData.append("facebook", "facebook");
  profileSocialData.append("linkedin", "linkedin");
  profileSocialData.append("youtube", "youtube");
  profileSocialData.append("instagram", "instagram");

  it("displays social inputs", () => {
    wrapper.find("#toggleSocials").simulate("click");
    expect(wrapper.find(SocialMedia).length).toEqual(1);
  });

  it("calls create profile fn with correct data", () => {
    wrapper
      .find(SocialMedia)
      .dive()
      .find("#facebook")
      .simulate("change", {
        target: { name: "facebook", value: "facebook" }
      });
    wrapper
      .find(SocialMedia)
      .dive()
      .find("#instagram")
      .simulate("change", {
        target: { name: "instagram", value: "instagram" }
      });
    wrapper
      .find(SocialMedia)
      .dive()
      .find("#youtube")
      .simulate("change", {
        target: { name: "youtube", value: "youtube" }
      });
    wrapper
      .find(SocialMedia)
      .dive()
      .find("#twitter")
      .simulate("change", {
        target: { name: "twitter", value: "twitter" }
      });
    wrapper
      .find(SocialMedia)
      .dive()
      .find("#linkedin")
      .simulate("change", {
        target: { name: "linkedin", value: "linkedin" }
      });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(mockCreateProfilefn.mock.calls[1][0]).toEqual(profileSocialData);
  });
});
