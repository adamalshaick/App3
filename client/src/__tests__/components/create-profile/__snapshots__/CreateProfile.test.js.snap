// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <Connect(Navbar) />
  <section
    className="entry container mt-md-5"
  >
    <div
      className="row"
    >
      <styled.div
        className="col-md-8 m-auto"
      >
        <styled.header
          className="text-center mb-4"
        >
          Create Your Profile
        </styled.header>
        <small
          className="d-block pb-3"
        >
          * = required fields
        </small>
        <form
          onSubmit={[Function]}
        >
          <ProfileInfo
            bio=""
            errors={Object {}}
            fileSelectedHandler={[Function]}
            handle=""
            location=""
            onChange={[Function]}
          />
          <div
            className="mb-3"
          >
            <button
              className="btn btn-outline-primary"
              id="toggleSocials"
              onClick={[Function]}
              type="button"
            >
              Add Social Media Links
            </button>
            <small
              className="text-muted d-block"
            >
              (Optional)
            </small>
          </div>
          <input
            className="btn btn-outline-primary float-right mt-4"
            type="submit"
            value="Submit"
          />
        </form>
      </styled.div>
    </div>
  </section>
</Fragment>
`;
